src/Tribe/Repositories/Event.php

```diff
@@ -1183,11 +1191,8 @@ protected function update_date_meta( array $postarr, $post_id = null ) {
 			);
 
 			// Let's make sure the timezone is always a string.
-			if ( isset( $postarr['meta_input']['_EventTimezone'] ) ) {
-				$postarr['meta_input']['_EventTimezone'] = Timezones::build_timezone_object(
-					$postarr['meta_input']['_EventTimezone']
-				)->getName();
-			}
+			$postarr['meta_input']['_EventTimezone'] = Timezones::build_timezone_object( $input_timezone )->getName();
```

Comment:
Wouldn't this create a problem of always saving a timezone? Which is a diff that than current behavior where we only modify if the timezone field is modified?

Reply:
Under the hood, we use the `update_meta_data` function that, for single custom fields like the `_EventTimezone` one, will not write the value to db if it's the same.

This example test will pass:
```php
public function should_not_write_the_same_value_twice() {
	$event         = static::factory()->event->create( [ 'meta_input' => [ '_EventTimezone' => 'America/New_York' ] ] );
	$updated_times = 0;

	add_action( 'update_postmeta', static function ( $meta_id, $object_id, $meta_key ) use ( &$updated_times ) {
		if ( $meta_key !== '_EventTimezone' ) {
			return null;
		}

		if ( ++ $updated_times > 1 ) {
			throw new AssertionFailedError( 'I should not be called twice' );
		}
	}, 10, 3 );

	update_post_meta( $event, '_EventTimezone', 'Europe/Paris' );
	update_post_meta( $event, '_EventTimezone', 'Europe/Paris' );
	update_post_meta( $event, '_EventTimezone', 'Europe/Paris' );

	$this->assertEquals( 1, $updated_times );
}
```